sql-parser/src/main/antlr/SqlBase.g4

```diff
@@ -237,6 +237,7 @@ primaryExpression
     | value=primaryExpression '[' index=valueExpression ']'                          #subscript
     | ident ('.' ident)*                                                             #dereference
     | primaryExpression CAST_OPERATOR dataType                                       #doubleColonCast
+    | timestamp=primaryExpression AT TIME ZONE zone=stringLiteralOrNull              #atTimezone
```

Comment:
Is `zone` supports only literals? In pg it can also be a column reference.

Reply:
My interpretation of https://www.postgresql.org/docs/current/functions-datetime.html#FUNCTIONS-DATETIME-ZONECONVERT where it says "In these expressions, the desired time zone zone can be specified either as a text string (e.g., 'America/Los_Angeles') or as an interval (e.g., INTERVAL '-08:00')." is that only literals and null are allowed.

Reply:
so no column names

Reply:
Seems like postgres allows any expressions:

```
select current_timestamp AT TIME ZONE (select '+3:00');
          timezone
----------------------------
 2019-08-06 12:04:11.132376
```

Given that we re-write it to a function that already supports arbritray expressions I'd be in favour of lifting this restriction. I don't like adding yet another expression variant with `stringLiteralOrNull`

Reply:
it is also not consistent with our `timezone` scalar, where zone can be an expression. so literal here is an artificial limitation.

Reply:
ok, on it